docs/specification/new-spec/blockchain.md

```diff
@@ -0,0 +1,389 @@
+A statement such as `x == y` is an assertion - if it fails, the item is invalid.
+
+We refer to certain globally available objects:
+`block` is the block under consideration,
+`prevBlock` is the `block` at the previous height,
+and `state` keeps track of the validator set, the consensus parameters
+and other results from the application.
+Elements of an object are accessed as expected,
+ie. `block.Header`. See [here](state.md) for the definition of `state`.
+
+## Header
+
+A Header is valid if its corresponding fields are valid.
+
+### Version
+
+Arbitrary string.
+
+### ChainID
+
+Arbitrary constant string.
+
+### Height
+
+```
+block.Header.Height > 0
+block.Header.Height == prevBlock.Header.Height + 1
+```
+
+The height is an incrementing integer. The first block has `block.Header.Height == 1`.
+
+### Time
+
+The median of the timestamps of the valid votes in the block.LastCommit.
+Corresponds to the number of nanoseconds, with millisecond resolution, since January 1, 1970.
+Increments with every new block.
+
+### NumTxs
+
+```
+block.Header.NumTxs == len(block.Txs)
+```
+
+Number of transactions included in the block.
+
+### TxHash
+
+```
+block.Header.TxHash == SimpleMerkleRoot(block.Txs)
+```
+
+Simple Merkle root of the transactions in the block.
+
+### LastCommitHash
+
+```
+block.Header.LastCommitHash == SimpleMerkleRoot(block.LastCommit)
+```
+
+Simple Merkle root of the votes included in the block.
+These are the votes that committed the previous block.
+
+The first block has `block.Header.LastCommitHash == []byte{}`
+
+### TotalTxs
+
+```
+block.Header.TotalTxs == prevBlock.Header.TotalTxs + block.Header.NumTxs
+```
+
+The cumulative sum of all transactions included in this blockchain.
+
+The first block has `block.Header.TotalTxs = block.Header.NumberTxs`.
+
+### LastBlockID
+
+```
+parts := MakeParts(block, state.ConsensusParams.BlockGossip.BlockPartSize)
+block.HeaderLastBlockID == BlockID{
+    SimpleMerkleRoot(block.Header),
+    PartsHeader{
+        SimpleMerkleRoot(parts),
+        len(parts),
+    },
+}
+```
+
+Previous block's BlockID. Note it depends on the ConsensusParams,
+which are held in the `state` and may be updated by the application.
+
+The first block has `block.Header.LastBlockID == BlockID{}`.
+
+### ResultsHash
+
+```
+block.ResultsHash == SimpleMerkleRoot(state.LastResults)
+```
+
+Simple Merkle root of the results of the transactions in the previous block.
+
+The first block has `block.Header.ResultsHash == []byte{}`.
+
+### AppHash
+
+```
+block.AppHash == state.AppHash
+```
+
+Arbitrary byte array returned by the application after executing and commiting the previous block.
+
+The first block has `block.Header.AppHash == []byte{}`.
+
+### ValidatorsHash
+
+```
+block.ValidatorsHash == SimpleMerkleRoot(state.Validators)
+```
+
+Simple Merkle root of the current validator set that is committing the block.
+This can be used to validate the `LastCommit` included in the next block.
```

Comment:
It might be useful adding a short description of the overall process (probably at the beginning of the document) so it's more clear the relation between blocks, and for example why in the current block we have commit from the previous block.